### DISCUSSION TITLE
Buffer not replaced when using vinegar-style API

### DISCUSSION THREAD

**Discussion by steliyan** [nvim-tree/nvim-tree.lua#2206] [Q&A]
Until now I've been using `open_replacing_current_buffer` which seems to be a legacy API (ref: https://github.com/nvim-tree/nvim-tree.lua/issues/1918). It was working as expected for long time, but recently it broke. More specifically, what is not working is selecting a file nvim-tree is opened.

I've tried the new one (`api.tree.toggle`), but still doesn't fit my use case.

I feel this is kinda related to this: https://github.com/nvim-tree/nvim-tree.lua/issues/1829, but I am not 100% sure.

Here are 2 screen captures which would be pretty self-explanatory. Note the example uses pretty simple file structure and used for a simple show-case, the real world example would have a lot of directory nesting.

I would like to have my nvim-tree buffer replaced when I choose a file outside nvim-tree. For the purpose of the video, I use Telescope, but similar behavior is observed if `:edit a.txt` command is used.

Am I missing something or should I file a bug?

# Reproduction

## `require'nvim-tree'.open_replacing_current_buffer()`

https://github.com/nvim-tree/nvim-tree.lua/assets/1850292/a1e526cc-af54-42f6-9231-d96f5069dc58

Here the buffer just flashes.

## `require'nvim-tree.api'.tree.toggle({current_window=true})`

https://github.com/nvim-tree/nvim-tree.lua/assets/1850292/3e809da0-99ca-4091-a92e-9b178b0dd319

Here a new buffer is created.

**Comment by alex-courtis** [comment]
I've not reviewed your entire post, however I think [Vinegar Style](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#vinegar-style) should meet your needs.

**Reply by steliyan** [reply]
I am using in Vinegar Style, my problem is I can't properly open a file outside nvim-tree when nvim-tree is the active buffer. If you check the videos, you'll see the problem. :)

**Reply by alex-courtis** [reply]
I see.

What happens when you use `api.node.open.replace_tree_buffer()` as per the wiki? `open_replacing_current_buffer` is not API.

Please post a [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml) with a ["Clean Room" Replication](https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting#clean-room-replication) so that we may replicate and fix this.

**Reply by steliyan** [reply]
If I use `replace_tree_buffer` (or the default `<C-e>` mapping), the file is edited in-place as expected.

I checked the code and I see that the window in the current tab page is closed and the selected file opened.

But if I do `:e some_file_name` then this is opened in a new window which was not the behavior before.

**Reply by alex-courtis** [reply]
> If I use `replace_tree_buffer` (or the default `<C-e>` mapping), the file is edited in-place as expected.

Good to hear that is working.

> But if I do `:e some_file_name` then this is opened in a new window which was not the behavior before.

That's hardcoded nvim-tree behaviour.

Please experiment with removing it, by commenting out the autocommand: https://github.com/nvim-tree/nvim-tree.lua/blob/498e8793bbe73ab5235b3ee8f0aee32f5d01649f/lua/nvim-tree.lua#L194

**Reply by alex-courtis** [reply]
Raised #2218 

[Pull requests](https://github.com/nvim-tree/nvim-tree.lua#contributing) are most gratefully appreciated!
